gemrb/GUIScripts/pst/GUIINV.py

```diff
@@ -406,36 +412,97 @@ def OnDragItem ():
 			GemRB.SetGlobal("APPEARANCE","GLOBAL",0)
 
 	UpdateInventoryWindow ()
+	return
+
+# Duplicated from InventoryCommon and should be removed once behaviour is consolidated
+def DecreaseStackAmount ():
+	"""Decreases the split size."""
+
+	Text = ItemAmountWindow.GetControl (6)
+	Amount = Text.QueryText ()
+	number = int ("0"+Amount)-1
+	if number<1:
+		number=1
+	Text.SetText (str (number))
+	return
 
+# Duplicated from InventoryCommon and should be removed once behaviour is consolidated
+def IncreaseStackAmount ():
+	"""Increases the split size."""
+
+	Text = ItemAmountWindow.GetControl (6)
+	Amount = Text.QueryText ()
+	number = int ("0"+Amount)+1
+	if number>StackAmount:
+		number=StackAmount
+	Text.SetText (str (number))
+	return
+
+# Duplicated from InventoryCommon and should be removed once behaviour is consolidated
 def DragItemAmount ():
 	"""Drag a split item."""
 
 	pc = GemRB.GameGetSelectedPCSingle ()
-	slot, slot_item = ItemHash[GemRB.GetVar ('ItemButton')]
-	Text = ItemAmountWindow.GetControl (6)
-	Amount = Text.QueryText ()
-	item = GemRB.GetItem (slot_item["ItemResRef"])
-	GemRB.DragItem (pc, slot, item["ItemIcon"], int ("0"+Amount), 0)
-	OpenItemAmountWindow ()
+
+	#emergency dropping
+	if GemRB.IsDraggingItem()==1:
+		GemRB.DropDraggedItem (pc, UsedSlot)
+		UpdateSlot (pc, UsedSlot-1)
+
+	slot_item = GemRB.GetSlotItem (pc, UsedSlot)
+
+	#if dropping didn't help, don't die if slot_item isn't here
+	if slot_item:
+		Text = ItemAmountWindow.GetControl (6)
+		Amount = Text.QueryText ()
+		item = GemRB.GetItem (slot_item["ItemResRef"])
+		GemRB.DragItem (pc, UsedSlot, item["ItemIcon"], int ("0"+Amount), 0)
+	OpenItemAmountWindow (None, UsedSlot)
+
 	return
 
-def OpenItemAmountWindow ():
-	global ItemAmountWindow
+def OpenItemAmountWindow (btn, slot):
+	"""Open the split window."""
+
+	global UsedSlot, OverSlot
+	global ItemAmountWindow, StackAmount
+
+	pc = GemRB.GameGetSelectedPCSingle ()
 
 	if ItemAmountWindow != None:
 		if ItemAmountWindow:
```

Comment:
this is completely redundant with the check above

Reply:
This function I barely touched, just in case it looked like it might be possible to merge back - which I think now it is, I was just stumped on something else so I didn't see it at first.

Reply:
I've changed `InventoryCommon`. This function is no longer used to close the window.

Reply:
No worries, I've always found that to be a bit counter-intuitive, it's odd how it can get confusing even when you know what it is doing

It turns out I don't need to clone any of this, it all folded in nicely with InventoryCommon with a slight tweak